drivers/sensor/bme680/bme680.c

```diff
@@ -346,13 +352,25 @@ static int bme680_read_compensation(const struct device *dev)
 static int bme680_init(const struct device *dev)
 {
 	struct bme680_data *data = dev->data;
-	const struct bme680_config *config = dev->config;
 	int err;
 
-	if (!device_is_ready(config->bus.bus)) {
-		LOG_ERR("I2C master %s not ready", config->bus.bus->name);
-		return -EINVAL;
+	err = bme680_bus_check(dev);
+	if (err < 0) {
+		LOG_ERR("Bus not ready for '%s'", dev->name);
+		return err;
+	}
+
+	/*
+	 * In case only I2C is used this saves a bit of code/time.
+	 * When using both (I2C and SPI) sensors it is safe to execute
+	 * this for the I2C device.
+	 */
+#if BME680_BUS_SPI
```

Comment:
Thanks for explaining. In that case I think it would be a little better to write it something like this:

```c
static inline bool bme680_is_on_spi(struct device *dev)
{
	const struct bme680_config *config = dev->config;

	return config->bus_io == &bme680_bus_io_spi;
}

/* ... */

#if BME680_BUS_SPI
	if (bme680_is_on_spi(dev)) {
		err = ...;
		if (err < 0) {
			return err;
		}
	}
#endif
```

IMO it is a bit better to have the I2C devices initialize the same way always, regardless of whether a SPI device happens to be in the same system. It may ease debugging in some cases and it definitely enhances consistency.

But it's not a blocking request; just please consider it.

Reply:
Thanks for taking another look at it and giving some more insight on that matter.

To have it consistent across all possible configurations wouldn't it make sense to 
remove the `#if BME680_BUS_SPI` directive aswell?

I have prepared the change as suggested by you (without the `#if`). If that makes sense to you I can push them.

Reply:
That's a fair point -- if it's safe to do regardless of bus type, might as well be consistent on all of them. Given the ifdef, I was assuming there was a reason to prefer not to do it on I2C, but if there isn't, then I'm fine either way.

Reply:
Well, it was probably me just over-complicating things -.-;
I pushed the changes, this way it is definitely easier on the eyes.

Reply:
...disregard that.. the directives are needed or else it won't build for I2C, obviously..
Sorry for the noise.